packages/grid/src/vaadin-grid-mixin.js

```diff
@@ -270,6 +270,33 @@ export const GridMixin = (superClass) =>
         reorderElements: true,
       });
 
+      // Chromium based browsers cannot properly generate drag images for elements
+      // that have children with massive heights. This workaround prevents crashes
+      // and performance issues by excluding the items from the drag image.
+      // https://github.com/vaadin/web-components/issues/7985
+      document.addEventListener(
+        'dragstart',
+        (e) => {
+          // The dragged element can be the element itself or a parent of the element
+          if (e.target !== this && !e.target.contains(this)) {
+            return;
+          }
+          // The threshold value 20000 provides a buffer to both
+          //   - avoid the crash and the performance issues
+          //   - unnecessarily avoid excluding items from the drag image
+          if (this.$.items.offsetHeight > 20000) {
+            const initialMaxHeight = this.$.items.style.maxHeight;
+            // Momentarily hides the items until the browser starts generating the
+            // drag image.
+            this.$.items.style.maxHeight = '0';
```

Comment:
For some reason that doesn't work properly for me. While the scroll container height reduces somewhat, it never goes to zero:

https://github.com/user-attachments/assets/582d85fb-55f7-46c5-99a5-ad9b24ce14eb

As an alternative I tried just hiding the items with `display: none`, which does seem to work:

https://github.com/user-attachments/assets/96194b9c-f706-40d9-bb6c-be949015d94d

Reply:
Using `display: none` would unfortunately lose the scroll position on drag

Reply:
Could setting `overflow: hidden` on the table element help?

Reply:
Indeed, if we can fix it without resetting that it would be better. Here is the repro if it helps:
```java
        Div target = new Div();
        target.setText("Drop here");
        target.getStyle().set("border", "1px solid black");
        target.getStyle().set("padding", "10px");

        DropTarget<Div> dropTarget = DropTarget.create(target);
        dropTarget.addDropListener(event -> {
            event.getDragSourceComponent().ifPresent(target::add);
        });
        add(target);

        Grid<String> grid = new Grid<>(String.class);
        ArrayList<String> list = new ArrayList<>();
        for (int i = 0; i < 1000; i++) {
            list.add(i + "");
        }
        grid.setItems(list);

        Div dragComp = new Div(grid);
        dragComp.setWidth("500px");
        dragComp.getStyle().setPadding("10px");
        dragComp.getStyle().setBackground("lightgray");
        add(dragComp);

        DragSource<Div> gridDragSource = DragSource.create(dragComp);
        gridDragSource.setDraggable(true);

        Div target2 = new Div();
        target2.setText("Or here");
        target2.getStyle().set("border", "1px solid black");
        target2.getStyle().set("padding", "10px");

        DropTarget<Div> dropTarget2 = DropTarget.create(target2);
        dropTarget2.addDropListener(event -> {
            event.getDragSourceComponent().ifPresent(target2::add);
        });

        add(target2);
```

Reply:
@vursen Chrome seems to ignore `overflow: hidden` when generating the drag image, or at least when calculating its size.

Reply:
The crash appears to be triggered by `transform: translateX(0)` on child elements. You can see this by adding and removing the transform property in the following example:

```html
<div draggable="true" style="width: 500px; height: 500px;">
  <div style="height: 100000px; background: red; transform: translateX(0)" ></div>
</div>
```

Reply:
Yeah, the issue occurs when the items container is forced into a separate layer, which is triggered by `transform: translate(0, 0)` and `position: sticky`:

<img width="1512" alt="image" src="https://github.com/user-attachments/assets/19905111-8e64-48f7-be80-5d62856739c1">

Reply:
> Could setting overflow: hidden on the table element help?

This (in additiona to the current fix) works well enough:
`this.$.table.style.overflow = 'hidden';`

https://github.com/user-attachments/assets/6a687c4e-d651-448a-9f2d-223f57640e18

Reply:
One fix for that issue is adding the following to the current fix:
```java
const gridRect = this.getBoundingClientRect();
e.dataTransfer.setDragImage(this, e.clientX - gridRect.left, e.clientY - gridRect.top);
```
But then any custom drag image set using `DragSource.setDragImage` will be disregarded. I could not find a better way to solve this or the reason why the wrong drag position occurs in the first place. This could maybe stated as a limitation for now. It avoids crashes and slowdowns, but does not display the custom drag image for massive grids. What are your opinions?

Reply:
I'm still curious, why do we need `transform` and `position: sticky` on the items element?

Reply:
> I'm still curious, why do we need transform and position: sticky on the items element?

One reason is that "scroll" event occurs asynchronously on Safari. If you have frozen columns on grid, you need to translate the cells on the columns for the frozen effect. On safari, if you scrolled left, you would see the frozen cells go a bit to the left also and only after the asynchronous "scroll" event, they would get translated to their expected positions. This would lead into a nasty glitch/jumpy behaviour when scrolling horizontally. So instead, we make the `<tbody>` element sticky to have it stay horizontally put while scrolling and then use javascript to move it and any possible frozen cells at the same time, synchronously, so there's no wobbliness.

Reply:
> Not a blocker – grid crash is worse.

So, the following seems to work both for a draggable grid/virtual list or a draggable container:

```js
const targetRect = e.target.getBoundingClientRect();
e.dataTransfer.setDragImage(e.target, e.clientX - targetRect.left, e.clientY - targetRect.top);
```

However, as @tomivirkki pointed out, it also affects the event beyond the scope of the grid. I am inclined towards not adding that part.